src/main/resources/io/viash/platforms/nextflow/arguments/_checkArgumentType.nf

```diff
@@ -106,10 +106,10 @@ def _checkArgumentType(String stage, Map par, Object value, String errorIdentifi
         // do nothing
       }
     }
-    if (value instanceof java.math.BigDecimal) {
+    if (value instanceof java.math.BigDecimal || value instanceof java.math.BigInteger) {
       value = value.doubleValue()
     }
-    if (value instanceof Float) {
+    if (value instanceof Float || value instanceof Integer || value instanceof Long) {
```

Comment:
Is `Short` and/or `Double` also a problem?

Reply:
I tried to resolve this by using a more idiomatic casting mechanism, namely:

```groovy
  // ...
  } else if (par.type == "double") {
    // cast to double if need be
    if (value !instanceof Double) {
      try {
        value = value as Double
      } catch (NumberFormatException e) {
        expectedClass = "Double"
      }
    }
  } // ...